### PR TITLE
Fix default language for email templates

### PR DESCRIPTION
**Pull request by Mixer57**
Added section 'DefaultLanguage' into AppSettings.json with Language property with default value as 'ru-RU'.
Added method GetEmailTemplateByLanguage into EmailFromTemplateService to search template with provided or default language.

### CODE REVIEW

File: src/OnlineSales/Configuration/AppSettings.cs [hunk 1]
```diff
@@ -122,4 +122,9 @@ public class CorsConfig
 public class EmailConfig : BaseServiceConfig
 {
     public bool UseSsl { get; set; }
+}
+
+public class DefaultLanguage
```
Comment (peterliapin) [inline src/OnlineSales/Configuration/AppSettings.cs:127]: @Mixer57 please move it into the existing ApiSettings section.

File: src/OnlineSales/Configuration/AppSettings.cs [hunk 2]
```diff
@@ -122,4 +122,9 @@ public class CorsConfig
 public class EmailConfig : BaseServiceConfig
 {
     public bool UseSsl { get; set; }
+}
+
+public class DefaultLanguage
+{
+    public string Language { get; set; } = "ru-RU";
```
Comment (peterliapin) [inline src/OnlineSales/Configuration/AppSettings.cs:129]: @Mixer57 the default language should be en-US.

File: src/OnlineSales/Services/EmailFromTemplateService.cs [hunk 3]
```diff
@@ -3,23 +3,37 @@
 // </copyright>
 
 using Microsoft.EntityFrameworkCore;
+using OnlineSales.Configuration;
 using OnlineSales.Data;
 using OnlineSales.DTOs;
 using OnlineSales.Entities;
 using OnlineSales.Infrastructure;
 using OnlineSales.Interfaces;
+using static Org.BouncyCastle.Math.EC.ECCurve;
```
Comment (peterliapin) [inline src/OnlineSales/Services/EmailFromTemplateService.cs:12]: @Mixer57 why do you need it?

File: src/OnlineSales/appsettings.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿{
   "AllowedHosts": "*",
   "MigrateOnStart": true,
-  "Plugins": [ "OnlineSales.Plugin.ReverseProxy", "OnlineSales.Plugin.EmailSync" ],
+  "Plugins": [ "OnlineSales.Plugin.ReverseProxy", "OnlineSales.Plugin.EmailSync", "OnlineSales.Plugin.XLTools" ],
```
Comment (peterliapin) [inline src/OnlineSales/appsettings.json:4]: @Mixer57 external plugins should never be included into the core configuration by default!

File: src/OnlineSales/Services/EmailFromTemplateService.cs [hunk 5]
```diff
@@ -51,11 +65,31 @@ private async Task<EmailTemplate> GetEmailTemplate(string name, int contactId)
...
+        private async Task<EmailTemplate?> GetEmailTemplateByLanguage(string name, string? language)
+        {
+            string defLang = config.Language!;
+
+            // set default if not set
+            language ??= defLang;
+
+            // try find template by provided language
+            // with 2 and 5 language codes
+            var template = await pgDbContext.EmailTemplates!.FirstOrDefaultAsync(x => x.Name == name && (x.Language.Length == 2 ? x.Language == language.Substring(0, 2) : x.Language == language));
```
Comment (peterliapin) [inline src/OnlineSales/Services/EmailFromTemplateService.cs:82]: @Mixer57 x.Language.Length should never be 2 letters, always 5. If it is not the case we need to fix it on the database level now.

**Review by peterliapin — CHANGES_REQUESTED** [review]

File: src/OnlineSales/appsettings.json [hunk 6]
```diff
@@ -126,7 +126,8 @@
   },
   "ApiSettings": {
     "MaxListSize": 10,
-    "MaxImportSize": "5MB"
+    "MaxImportSize": "5MB",
+    "Language": "en-US"
```
Comment (peterliapin) [inline src/OnlineSales/appsettings.json:130]: Please rename to DefaultLanguage

File: src/OnlineSales/Configuration/AppSettings.cs [hunk 7]
```diff
@@ -71,6 +71,8 @@ public class ApiSettingsConfig
     public int MaxListSize { get; set; }
 
     public string MaxImportSize { get; set; } = string.Empty;
+
+    public string Language { get; set; } = "ru-RU";
```
Comment (peterliapin) [inline src/OnlineSales/Configuration/AppSettings.cs:75]: The default should be en-US

**Review by peterliapin — CHANGES_REQUESTED** [review]

File: src/OnlineSales/Services/EmailFromTemplateService.cs [hunk 8]
```diff
@@ -15,11 +16,23 @@ public class EmailFromTemplateService : IEmailFromTemplateService
...
         private readonly PgDbContext pgDbContext;
+        private readonly ApiSettingsConfig config;
 
-        public EmailFromTemplateService(IEmailWithLogService emailWithLogService, PgDbContext pgDbContext)
+        public EmailFromTemplateService(IEmailWithLogService emailWithLogService, PgDbContext pgDbContext, IConfiguration configuration)
         {
             this.emailWithLogService = emailWithLogService;
             this.pgDbContext = pgDbContext;
+
+            var settings = configuration.GetSection("ApiSettings").Get<ApiSettingsConfig>();
```
[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Services/EmailFromTemplateService.cs:26]: IOptions<ApiSettingsConfig> apiSettingsConfig used instead configuration.GetSection("ApiSettings").Get<ApiSettingsConfig>();

File: src/OnlineSales/appsettings.json (hunk 6 above)
Comment (Mixer57) [reply src/OnlineSales/appsettings.json:130]: Renamed

**Pull request merged**